src/components/ButtonsOnPetPage.tsx

```diff
@@ -1,16 +1,53 @@
+import { useState } from 'react';
 import { Button } from '@/components/ui/button';
 import { TbEdit } from 'react-icons/tb';
 import { TbDotsVertical } from 'react-icons/tb';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { DeletePetModal } from '@/components/modals/DeletePetModal';
+import { useRouter } from 'next/navigation';
+import { removePet } from '@/data/pets/removePet';
+
+export const ButtonsOnPetPage: React.FC<PetData> = ({ pet_id, pet_name }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const router = useRouter();
+
+  const handleDeletePet = () => {
+    removePet(pet_id);
+    setIsModalOpen(false);
+    router.push('/user');
```

Comment:
🔴  Update: Changed router.push to router.replace to prevent adding an unnecessary entry in the navigation history during redirection.

Reason: When navigating to the single pet data tab and subsequently removing a pet, the app redirects to the list of all pets. However, using router.push creates an additional history entry. This causes an issue where, if the user goes back in the browser history, they are taken back to the pet data tab of the removed pet. By switching to router.replace, this issue is resolved as no additional history entry is created during the redirect.

Reply:
Great catch! I just realzied the difference between `router.push` and `router.replace`, here it's well explaind: https://stackoverflow.com/questions/52954837/difference-between-react-router-push-and-router-replace